src/OpenCL.cpp

```diff
@@ -512,11 +514,13 @@ void OpenCL_Network::convolve(int filter_size, int channels, int outputs,
         m_convolve_kernel = &opencl_thread_data.m_convolve1_kernel;
     }
 
-    // Input channel grouping
+    // Input channel grouping in multiples of 8
     int channelGroup = 8;
     int channelShift = 3;
+
     // Input layer is not a multiple of 8
-    if (channels == 18) {
+    if (channels % 8 != 0) {
```

Comment:
I thought this was slightly better semantically but I can revert if you like the hard coded 18 (or I can replace with Network::INPUT_CHANNELS)

Reply:
It's fine.